### PR TITLE
utilty to summarize instance usages for various cloud vendors

### PR DESCRIPTION
**Pull request by pruan-rht**
usage is `./tools/cloud_cop.rb aws` to get summary of all regions or `./tools/cloud_cop.rb aws --region=us-east-2` to just get the summary for that region.  Currently only support AWS, will add other cloud providers later.  Here's an example output
http://pastebin.test.redhat.com/804638

### CODE REVIEW

**Comment by liangxia** [comment]
Great. I like it.

**Comment by pruan-rht** [comment]
@akostadinov this is ready for review now.  Thanks!

**Comment by pruan-rht** [comment]
@akostadinov bump...

**Comment by pruan-rht** [comment]
@akostadinov , bump, bump. :)

File: lib/launchers/openstack10.rb [hunk 1]
```diff
@@ -822,6 +824,34 @@ def get_routers
...
+      raise res[:response] unless res[:success]
+      return res[:parsed]
+    end
+
+    def get_running_instances
+      threads = []
+      running_inst = {}
+      instances = get_objects("servers")
+      instances.each do |instance|
+        threads << Thread.new(instance) do | i|
```
Comment (akostadinov) [inline lib/launchers/openstack10.rb]: better throttle this to 15 or so connections

Comment (akostadinov) [reply lib/launchers/openstack10.rb]: use method `array#each_slice`

File: lib/instance_summary.rb [hunk 2]
```diff
@@ -0,0 +1,287 @@
+require 'text-table'
+require 'thread'
+require 'thwait'
+
+module BushSlicer
+  class InstanceSummary
```
Comment (akostadinov) [inline lib/instance_summary.rb]: Could you add some description about what this class is doing? I looks like belonging to `tools/lib`.

File: lib/instance_summary.rb [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    attr_accessor :jenkins
+
+    def initialize(jenkins: nil)
```
Comment (akostadinov) [inline lib/instance_summary.rb]: you allow initialization without a jenkins instance. But later you call methods on `jenkins` without checking for `nil`. So perhaps `: nil` part needs to be removed. To make sure objects are always created with a `jenkins` object set.

File: lib/jenkins.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+require 'net/https'
+require 'thread'
+
+module BushSlicer
+  class Jenkins
+    attr_accessor :client, :build_map
+    def initialize
+      @client = JenkinsApi::Client.new(:server_url => 'https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com',
```
Comment (akostadinov) [inline lib/jenkins.rb]: please do not hardcode this. it needs to be a config under `private/config/config.yaml`. Actually it is already there under ```
services:
  jenkins: ...
```

File: lib/jenkins.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
+require 'net/https'
+require 'thread'
+
+module BushSlicer
+  class Jenkins
```
Comment (akostadinov) [inline lib/jenkins.rb]: This looks like more suitable for `tools/lib`.

File: lib/launchers/openstack10.rb (hunk 1 above)
Comment (pruan-rht) [reply lib/launchers/openstack10.rb]: @akostadinov what do you mean by using #each_slice?   We probably don't need to throttle as Jenkins only returns 100 entries max.  Also, if we do want to use threadpool, then that's another gem we need to install

Comment (akostadinov) [reply lib/launchers/openstack10.rb]: 100 connections at the same time not very little but sounds not terrible.
wrt a thread pool, we can jst create that number of threads per each slice. Not needed to use a thread pool.

File: lib/jenkins.rb (hunk 4 above)
Comment (pruan-rht) [reply lib/jenkins.rb]: @akostadinov  how do I get the `conf`

**Pull request merged**